File: easy/1_Two_Sum/main.c
#include <stdlib.h>

/* Way 1: Brute Forced, O(n^2) */
int* twoSum1(int* nums, int numsSize, int target, int* returnSize) {
    for (int i = 0; i < numsSize; i++) {
        for (int j = i + 1; j < numsSize; j++) {
            if (nums[i] + nums[j] == target) {
                int* result = (int*)malloc(2 * sizeof(int));

                if (result != NULL) {
                    result[0] = i;
                    result[1] = j;
                    *returnSize = 2;
                    return result;

                } else {
                    *returnSize = 0;
                    return NULL;
                }
            }
        }
    }

    *returnSize = 0;
    return NULL;
}

File: easy/1_Two_Sum/main.py
from typing import List

class Solution:
    # Way 1: Brute Force, O(n^2)
    def twoSum1(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                if(nums[i] + nums[j] == target):
                    return [i, j]

    # Way 2: Hash Map, O(n)
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen : map = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if(nums[i] in seen):
                return [seen[nums[i]], i]
            else:
                seen[diff] = i